src/ast/nodes.hpp

```diff
@@ -112,12 +112,13 @@ struct ReturnStatement : public AstNode<ReturnStatement> {
   ::std::unique_ptr<Expression> expression;
 
   void generate(CodeGeneratorEvaluator &_evaluator) const {
+    // TODO: Instead of loadValueInRegister, _evaluator.loadValueInRegister(...) would be better
```

Comment:
Nice, interesting thought! Why would it be better? You would have to give expr to evaluator, and evaluator would then need to be aware of what an expr is right?